### PR TITLE
remove duplication from normalize entity name

### PR DESCRIPTION
**Pull request by tyleriguchi**
Let me know if this looks alright, first time contributing. In reference to #3730

### CODE REVIEW

File: lib/utilities/valid-component-name.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+var SilentError = require('../errors/silent');
+
+module.exports = function(entityName) {
```
Comment (stefanpenner) [inline lib/utilities/valid-component-name.js]: can you add an explicit test for this new extracted module?

basically for the two branches:

* test that it throws, with the correct error type and message
* test that it returns entityName

**Comment by trabus** [comment]
>  first time contributing

@tyleriguchi nice work! Let me know if you need any help with the tests.

**Comment by tyleriguchi** [comment]
Added tests to confirm it throws an error and message, and that it returns an entity name

**Comment by stefanpenner** [comment]
Thank you kindly

**Pull request merged**